### DISCUSSION TITLE
issue: Error display does to update

### DISCUSSION THREAD

**Discussion by zfreiter** [react-hook-form/react-hook-form#10259] [Q&A] [answered]
### Version Number

7.43.8

### Codesandbox/Expo snack

https://codesandbox.io/s/quizzical-knuth-p229bv

### Steps to reproduce

1. Add a project name in the project field.
2. Hit submit.
3. Error shows, "Please add one query to create a project."
4. Select add query from any three query buttons.
5. Enter in a value for selected query.
6. Error continues to show, "Please add one query to create a project."
7. Select submit.

### Expected behaviour

In step 5, "Enter in a value for selected query", when a value is entered the error should disappear.
The yup validation custom createError is displaying to the console that everything is correct, so I am unsure why the error does not disappear. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
```
resolver: async (data, context, options) => {
  // you can debug your validation schema here
  console.log('formData', data)
  console.log('validation result', await anyResolver(schema)(data, context, options))
  return anyResolver(schema)(data, context, options)
},
```

please run the above code snippet to debug the root cause.

**Reply by zfreiter** [reply]
Thank you. Using trigger fixed the issue, but using the debugging method above gave me a better insight into handling issues like this in the future.

**Comment by hekatx** [comment]
I'm guessing validation for `projectName` is not running because it doesn't need to (the field hasn't changed). Calling `trigger("projectName");` on each `Add [provider] query` button seems to do the trick. 

I think even the docs recommend using `trigger` when the validation of certain field is dependent on the state of other fields: https://react-hook-form.com/api/useform/trigger/

**Reply by zfreiter** [reply, marked as answer]
Thank you that fixed the issue.
